pyro/poutine/util.py

```diff
@@ -128,3 +128,40 @@ def all_escape(trace, msg):
     return (msg["type"] == "sample") and \
         (not msg["is_observed"]) and \
         (msg["name"] not in trace)
+
+
+def broadcast_enum_filter(msg):
```

Comment:
It seems like these helpers would naturally live in pyro.infer.enum, where most of the enumeration logic lives.